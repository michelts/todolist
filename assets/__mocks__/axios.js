export default {
  get: jest.fn(),
  post: jest.fn(),
  put: jest.fn(),
};
