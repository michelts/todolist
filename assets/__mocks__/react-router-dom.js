export const useHistory = jest.fn();

export const Link = () => null;
