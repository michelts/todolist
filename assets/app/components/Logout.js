import React from 'react';
import axios from 'axios';
import { useHistory } from 'react-router-dom';
import Loading from 'common/components/Loading';

const Logout = () => {
  const { push } = useHistory();

  React.useEffect(() => {
    axios.get('/api/v1/users/logout/').then(() => push('/'));
  }, []);

  return (
    <Loading />
  );
}

export default Logout;
