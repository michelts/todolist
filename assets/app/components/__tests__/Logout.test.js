import React from 'react';
import { shallow } from 'enzyme';
import { useHistory } from 'react-router-dom';
import axios from 'axios';
import Logout from '../Logout';

const getWrapper = () => {
  const push = jest.fn();
  useHistory.mockReturnValue({ push });
  const wrapper = shallow(<Logout />);
  return { wrapper, history: { push } };
};

describe('Logout component', () => {
  beforeEach(() => {
    jest.spyOn(React, 'useEffect').mockImplementation((effect) => effect());
    axios.get.mockResolvedValue({});
  });

  it('should logout user on mount and redirect him to home', async () => {
    const { history } = getWrapper();
    await axios;
    expect(history.push).toHaveBeenCalledWith('/');

    // Ensure it was called twice (enzyme don't work well with useEffect)
    expect(React.useEffect).toHaveBeenCalledWith(expect.anything(), []);
  });

  it('should render a loading component while processing logout', () => {
    const { wrapper } = getWrapper();
    expect(wrapper).toMatchSnapshot();
  });
});
