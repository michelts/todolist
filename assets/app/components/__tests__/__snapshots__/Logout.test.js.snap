// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout component should render a loading component while processing logout 1`] = `<Loading />`;
